packages/http-client-python/generator/pygen/codegen/models/code_model.py

```diff
@@ -415,7 +415,7 @@ def license_header(self) -> str:
             license_header = self.yaml_data.get("licenseInfo", {}).get("header", "")
         else:
             # typespec azure case without custom license and swagger case
-            license_header = self.options.get("header_text", DEFAULT_HEADER_TEXT)
+            license_header = self.options.get("header_text") or DEFAULT_HEADER_TEXT
```

Comment:
options is [built](https://github.com/microsoft/typespec/blob/8eabb0750a421f2ae4de0cf61173d6114ea7e5cf/packages/http-client-python/generator/pygen/codegen/__init__.py#L306) from [OptionsRetriever](https://github.com/microsoft/typespec/blob/8eabb0750a421f2ae4de0cf61173d6114ea7e5cf/packages/http-client-python/generator/pygen/codegen/__init__.py#L26) which always have value of key even if it is None: https://github.com/microsoft/typespec/blob/8eabb0750a421f2ae4de0cf61173d6114ea7e5cf/packages/http-client-python/generator/pygen/codegen/__init__.py#L52-L54

Reply:
The fix is to make sure SDK from swagger or azure always have license header.